src/main/java/com/faforever/client/map/MapService.java

```diff
@@ -412,6 +413,21 @@ public Path getPathForMap(String technicalName) {
   }
 
 
+  public Path getPathForMapInsensitive(String approxName) {
+    String normalizedName = approxName.toLowerCase();
+    try (DirectoryStream<Path> stream = Files.newDirectoryStream(customMapsDirectory)) {
+      for (Path entry : stream) {
+        if (entry.getFileName().toString().toLowerCase().equals(normalizedName)) {
+          return entry;
+        }
+      }
+    } catch (IOException e) {
```

Comment:
lets simply not catch it at all, returning nul only delays the issue and probably results in a NPE, just use 

```suggestion
noCatch(()->{
     //Your code
});
```

Reply:
Some code actually uses null as a missing map indicator, hm...

Reply:
ok then